lib/src/values/values.dart

```diff
@@ -11,3 +13,61 @@ part 'logic_value.dart';
 part 'small_logic_value.dart';
 part 'big_logic_value.dart';
 part 'filled_logic_value.dart';
+
+/// Allows random generation of [LogicValue] for [BigInt] and [int].
+extension RandLogicValue on Random {
+  /// Generate non-negative random [BigInt] value that consists of
+  /// [numBits] bits.
+  ///
+  /// Example:
+  ///
+  /// ```dart
+  /// // generate 100 bits of random BigInt
+  /// final bigInt = Random(10).nextBigInt(numBits: 100);
+  /// ```
+  BigInt _nextBigInt({required int numBits}) {
+    var result = BigInt.zero;
+    for (var i = 0; i < numBits; i += 32) {
+      result = (result << 32) | BigInt.from(nextInt(1 << 32));
+    }
+    return result & ((BigInt.one << numBits) - BigInt.one);
+  }
+
+  /// Generate non-negative random [LogicValue] based on [width] and [max] num.
+  /// The random number can be mixed in invalid bits x and z by set
+  /// [hasInvalidBits] to `false`.
+  ///
+  /// Example:
+  ///
+  /// ```dart
+  /// // generate 100 bits of random BigInt
+  /// final bigInt = Random(10).nextBigInt(numBits: 100);
+  /// ```
+  LogicValue nextLogicValue({
+    required int width,
+    int? max,
+    bool hasInvalidBits = false,
```

Comment:
I think it might be better to call this `includeInvalidBits` since we're including it, it doesn't already have it?